tests/erc721/erc721_test.go

```diff
@@ -162,6 +171,289 @@ func TestOpenSeaProxyApproval(t *testing.T) {
+			mints:        true,
+			hasProxy:     true,
+			wantApproved: true,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+			},
+		},
+		{
+			name:         "opt out with proxy after mint",
+			mints:        true,
+			hasProxy:     true,
+			wantApproved: false,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: false,
+				},
+			},
+		},
+		{
+			name:         "opt in with proxy",
+			mints:        false,
+			hasProxy:     true,
+			wantApproved: true,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+			},
+		},
+		{
+			name:         "opt out with proxy",
+			mints:        false,
+			hasProxy:     true,
+			wantApproved: false,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: false,
+				},
+			},
+		},
+		{
+			name:         "opt in after mint",
+			mints:        true,
+			hasProxy:     false,
+			wantApproved: true,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+			},
+		},
+		{
+			name:         "opt out after mint",
+			mints:        true,
+			hasProxy:     false,
+			wantApproved: false,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: false,
+				},
+			},
+		},
+		{
+			name:         "opt in",
+			mints:        false,
+			hasProxy:     false,
+			wantApproved: true,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: true,
+				},
+			},
+		},
+		{
+			name:         "opt out",
+			mints:        false,
+			hasProxy:     false,
+			wantApproved: false,
+			wantEvents: []*ERC721ApprovalForAll{
+				{
+					Owner:    sim.Addr(tokenReceiver),
+					Operator: sim.Addr(proxy),
+					Approved: false,
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sim, nft, filter := deploy(t)
+			start := sim.BlockNumber().Uint64()
+
+			if tt.hasProxy {
+				openseatest.SetProxyTB(t, sim, sim.Addr(tokenReceiver), sim.Addr(proxy))
+			}
+
+			if tt.mints {
+				sim.Must(t, "Mint(%d)", notExists)(nft.Mint(sim.Acc(tokenReceiver), big.NewInt(notExists)))
+			}
+
+			sim.Must(t, "%T.setApprovalForAll(proxy,%v)", nft, tt.wantApproved)(
+				nft.SetApprovalForAll(sim.Acc(tokenReceiver), sim.Addr(proxy), tt.wantApproved),
```

Comment:
Please use different fields for the action and for the expected result. Even if they're always the same value, it makes it much clearer what the intended actions will be when just scanning the table.